### DISCUSSION TITLE
Bird model

### DISCUSSION THREAD

**Discussion by sergewich** [weecology/DeepForest#616] [Q&A] [answered]
Hi, 
Does the bird model also run in R? I am not sure how to select it in R? Does anyone have any experience with using that model on orthomosaics in R? Is that even possible or can it only be run on single images?
Serge

**Comment by ethanwhite** [comment]
Hi @sergewich!

It's definitely possible to run the bird model on orthomosaics, that's one of our primary personal use cases and as a result of the lot the training data is from orthos as well. The big question is probably whether this is working in R since we've had a number of issues on and off with keeping the R wrapper working across operating systems.

Here are the instructions from predicting on an ortho in the R wrapper: https://github.com/weecology/deepforestr?tab=readme-ov-file#predict-a-tile. If this isn't working for you can you let us know what operating system you're using and show us the error.

**Reply by sergewich** [reply]
Hi, 
Thank you. I will give that a go. Otherwise I assume it works in Python and I can see if someone can help with that. How do we call in the bird model into R? That is not completely clear to me from the R code.

**Reply by ethanwhite** [reply]
Yes, it definitely works in Python. We run it frequently. See the [Getting Started](https://deepforest.readthedocs.io/en/latest/getting_started.html) page, down through [Predict a tile](https://deepforest.readthedocs.io/en/latest/getting_started.html#predict-a-tile) and just replace:

```
model = main.deepforest()
model.use_release()
```

with

```
model = main.deepforest(label_dict={"Bird":0})
model.use_bird_release()
```

**Comment by bw4sz** [comment]
Keep us updated here, I have been working on a bird model tutorial that hasn't been merged into docs. what's your use case?

**Reply by sergewich** [reply]
Great.  For now I would like to try it out for bird counts on islands in the caribbean that have been done with drones. 
Unfurtunately I run into an error:
C:\Users\serge\anaconda3\DeepForest>conda env create --file=environment.yml

CondaValueError: invalid package specification: setuptools <59.6.

I am afraid I do not know how to solve that error. Any thoughts?

**Reply by ethanwhite** [reply]
Thanks @sergewich. Is this coming from one of the R installation commands or is this a command that you are running yourself at the Windows terminal? Or something else?

**Comment by sergewich** [comment]
Sorry, I should have made that clear. I am trying to install package in Python so am running the windows command or anaconda prompt. I tried the various installation options that you suggest on Github and all of them run into issues with either solving environment or pywheels. I realise it is extremely difficult to troubleshoot from afar. I have a colleague who works a lot with Python so plan to ask for his help potentially if need be. But any thoughts that you have on how to solve this are very welcome as I am keen to get this working.

**Reply by ethanwhite** [reply, marked as answer]
Thanks. So this doesn't work for you?

```
pip install deepforest
```

Assuming that let's try a standard install into a conda environment. Open the Anaconda prompt and try running the following:

```
conda create -n deepforestenv python=3.11
conda activate deepforestenv
pip install deepforest
```

**Comment by sergewich** [comment]
Many thanks. Just this morning I managed to install it but with Python 
3.9. Should I update this to 3.11? If so I need to see how I do that.

I am running into issues when loading the sample image OSBS_029.png.

TypeError                                 Traceback (most recent call last)
Cell In[11], line 1----> 1 img =model.predict_image(sample_image_path,return_plot=True)

File ~\anaconda3\envs\deepforest\lib\site-packages\deepforest\main.py:334, in deepforest.predict_image(self, image, path, return_plot, color, thickness)    332# sanity checks on input images    333ifnottype(image) == np.ndarray:
--> 334raiseTypeError("Input image is of type {}, expected numpy, if reading "    335"from PIL, wrap in "    336"np.array(image).astype(float32)".format(type(image)))
     338if image.dtype !="float32":
     339     warnings.warn(f"Image type is {image.dtype}, transforming to float32. "    340f"This assumes that the range of pixel values is 0-255, as "    341f"opposed to 0-1.To suppress this warning, transform image "    342f"(image.astype('float32')")

TypeError: Input image is of type <class 'str'>, expected numpy, if reading from PIL, wrap in np.array(image).astype(float32)

I will try to figure out why that is. Maybe the image is not loading? I tried with my own images and get the same issue.



------ Original Message ------
From: "Ethan White" ***@***.***>
To: "weecology/DeepForest" ***@***.***>
Cc: "sergewich" ***@***.***>; "Mention" 
***@***.***>
Sent: 07/03/2024 12:05:01
Subject: Re: [weecology/DeepForest] Bird model (Discussion #616)

**Reply by ethanwhite** [reply]
Trying adding an explicit name to the first argument like shown at the top of the [getting started page](https://deepforest.readthedocs.io/en/latest/getting_started.html):

```python
 img = model.predict_image(path=sample_image_path, return_plot=True)
```

This function can take either an actual image or a path as input and I think the image itself is the first argument so you need to name the argument to use the path instead.

**Comment by sergewich** [comment]
Super, it works now. Many thanks. I will try the bird model as well and 
let you know how it goes.

------ Original Message ------
From: "Ethan White" ***@***.***>
To: "weecology/DeepForest" ***@***.***>
Cc: "sergewich" ***@***.***>; "Mention" 
***@***.***>
Sent: 07/03/2024 13:00:42
Subject: Re: [weecology/DeepForest] Bird model (Discussion #616)

**Comment by sergewich** [comment]
Things go fine with the single images but with the raster files for orthos I am struggling a bit. I am getting some error messages and I am not sure what they are. Also I am not sure how to see the boxes on an ortho or is that not possible? I attached a screenshot in case you have time to take a look.
![Screenshot (110)](https://github.com/weecology/DeepForest/assets/52251953/3f0724a6-1d0d-4efe-b89e-5fe769db31c5)

**Reply by sergewich** [reply]
I think there was something with the raster as I processed it again and now it works fine.

**Comment by ethanwhite** [comment]
Glad things are working for you now! If you're in good shape feel free to go ahead and close this. You can always open a new issues/discussion if you have more questions. Also, definitely let us know how things go with the results!
